### PR TITLE
Integration test for network access of runner

### PR DESCRIPTION
**Pull request by yhaliaw**
### Overview

Add integration test to test whether web service on the juju machine can be access by the LXD instance hosting the runners.

### Rationale

Job on the runner should not access any service on the juju machine that is not intended for them.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by cbartz — APPROVED** [review]
LGTM

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -172,6 +173,7 @@ async def run_in_unit(unit: Unit, command: str, timeout=None) -> tuple[int, str
     Args:
         unit: Juju unit to execute the command in.
         command: Command to execute.
+        timeout: Amount of time to wait for the execution.
```
Comment (arturo-seijas) [inline tests/integration/helpers.py:176]: How come this passed linting ealier?

Comment (cbartz) [reply tests/integration/helpers.py:176]: `pydocstyle` is configured to only go over `src`.

Comment (arturo-seijas) [reply tests/integration/helpers.py:176]: Makes sense, I suppose

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**